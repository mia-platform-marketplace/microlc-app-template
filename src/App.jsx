import React from 'react'
import {Routes, Route, Link} from 'react-router-dom'

import logo from './logo.svg'
import './App.css'

function App () {
  return (
    <div className='App'>
      <Routes>
        <Route
          element={
            <header className='App-header'>
              <img alt='logo' className='App-logo' src={logo} />
              <Link
                className='App-link'
                to='/about'
              >
                <p>{'Go To About Page'}</p>
              </Link>
            </header>
          }
          path='/'
        ></Route>
        <Route
          element={
            <header className='App-header'>
              <div style={{fontSize: '40px', fontWeight: 700}}>{'About'}</div>
              <Link
                className='App-link'
                to='/'
              >
                <p>{'Go Home'}</p>
              </Link>
            </header>
          }
          path='/about'
        ></Route>
      </Routes>
    </div>
  )
}

export default App
