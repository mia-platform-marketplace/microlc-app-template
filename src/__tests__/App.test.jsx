import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import {render} from '@testing-library/react'

import App from '../App'

function renderWrapperContainer (children, initialEntries) {
  return render(
    <MemoryRouter initialEntries={initialEntries}>
      {children}
    </MemoryRouter>
  )
}

it('should render', () => {
  const {asFragment} = renderWrapperContainer(<App />)
  expect(asFragment()).toMatchSnapshot()
})
