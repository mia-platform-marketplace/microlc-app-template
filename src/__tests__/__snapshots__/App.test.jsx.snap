// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <a
        class="App-link"
        href="/about"
      >
        <p>
          Go To About Page
        </p>
      </a>
    </header>
  </div>
</DocumentFragment>
`;
